pkg/reconciler/pipelinerun/metrics.go

```diff
@@ -134,9 +134,13 @@ func (r *Recorder) DurationAndCount(pr *v1alpha1.PipelineRun) error {
 		status = "failed"
 	}
 
+	pipelineName := "anonymous"
+	if pr.Spec.PipelineRef != nil && pr.Spec.PipelineRef.Name != "" {
+		pipelineName = pr.Spec.PipelineRef.Name
+	}
 	ctx, err := tag.New(
 		context.Background(),
-		tag.Insert(r.pipeline, pr.Spec.PipelineRef.Name),
```

Comment:
The failure was legit :sweat_smile: this was putting the controller in a crashloop :stuck_out_tongue_closed_eyes: